### PR TITLE
Update finetune label models

### PR DESCRIPTION
**Pull request by nithinraok**
# What does this PR do ?

Upgraded speaker recognition fine-tune script

**Collection**: [ASR - Speaker Recognition]

# Changelog 
- Removed set_up_finetune_model definition in label_models
- Updated speaker_reco_finetune.py script, to resemble PTL training method
- Now, fine-tune can be directly used from config. Added titanet-finetune.yaml config
- Updated notebook tutorials for the fine-tuning section

# Usage
* You can potentially add a usage example below

```bash
python speaker_reco_finetune.py 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [x] Documentation

### CODE REVIEW

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 1]
```diff
@@ -1,1267 +1,1228 @@
...
+   "source": [
+    "# Training\n",
+    "Import necessary packages"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note: All the following steps are just for explanation of each section, but one can use provided [training script](https://github.com/NVIDIA/NeMo/blob/main/examples/speaker_tasks/recognition/speaker_reco.py) to train the model from command line."
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:279]: use provided -> use the provided
to train the model from command line -> to launch training in the command line

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 2]
```diff
@@ -1,1267 +1,1228 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "KyDVdtjAL2__"
+   },
+   "source": [
+    "As the goal of most speaker-related systems is to get good speaker level embeddings that could help distinguish from\n",
+    "other speakers, we shall first train these embeddings in end-to-end\n",
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:259]: in end-to-end -> in an end-to-end

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 3]
```diff
@@ -1,1267 +1,1228 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "Or8g1cksSf8C"
+   },
+   "source": [
+    "As any NeMo model is inherently a PyTorch Lightning Model, it can easily be trained in a single line - trainer.fit(model)!\n",
+    "We see below that the model begins to get modest scores on the validation set after just 5 epochs of training"
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:590]: We see below that -> Below we see that

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 4]
```diff
@@ -1,1267 +1,1228 @@
...
+    "## Saving and Restoring via PyTorch Lightning Checkpoints\n",
+    "When using NeMo for training, it is advisable to utilize the exp_manager framework. It is tasked with handling checkpointing and logging (Tensorboard as well as WandB optionally!), as well as dealing with multi-node and multi-GPU logging.\n",
+    "\n",
+    "Since we utilized the exp_manager framework above, we have access to the directory where the checkpoints exist.\n",
+    "\n",
+    "exp_manager with the default settings will save multiple checkpoints for us -\n",
+    "\n",
+    "1) A few checkpoints from certain steps of training. They will have --val_loss= tags\n",
+    "\n",
+    "2) A checkpoint at the last epoch of training denotes by --last.\n",
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:672]: (denote should be passive voice)
A checkpoint at the last epoch of training denotes by --last. ->
Checkpoints at the last epoch of training are denoted by --last.

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 5]
```diff
@@ -1,1267 +1,1228 @@
...
+    "2. Finetuning from a PTL checkpoint. \n",
+    "\n",
+    "Since finetuning from a large pretrained model is more common, I shall use it to demostrate finetuning procedure. In order to make finetuning step independent from training from scratch, we use another config. Here we shall use `titanet-finetune.yaml` config, that is created to show finetuning on pretrained titanet-large model. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note: You may use [finetune-script](https://github.com/NVIDIA/NeMo/blob/main/examples/speaker_tasks/recognition/speaker_reco_finetune.py) to train from command line directly. Following is just a demonstration of the script"
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:762]: to train from command line -> launch training in the command line

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 6]
```diff
@@ -1,1267 +1,1228 @@
...
+   "metadata": {
+    "colab_type": "text",
+    "id": "kV9gInFwQ2F5"
+   },
+   "source": [
+    "There are couple of ways we can finetune a speaker recognition model. \n",
+    "1. Finetuning using a pretrained model published on NGC. \n",
+    "2. Finetuning from a PTL checkpoint. \n",
+    "\n",
+    "Since finetuning from a large pretrained model is more common, I shall use it to demostrate finetuning procedure. In order to make finetuning step independent from training from scratch, we use another config. Here we shall use `titanet-finetune.yaml` config, that is created to show finetuning on pretrained titanet-large model. "
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:755]: demostrate -> demonstrate

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 7]
```diff
@@ -1,1267 +1,1228 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "kV9gInFwQ2F5"
+   },
+   "source": [
+    "There are couple of ways we can finetune a speaker recognition model. \n",
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:751]: There are couple of ways  -> There are a couple of ways

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 8]
```diff
@@ -1,1267 +1,1228 @@
...
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "o1ojB0cZMSmv"
+   },
+   "outputs": [],
+   "source": [
+    "import nemo\n",
+    "# NeMo's ASR collection - this collections contains complete ASR models and\n",
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:293]: this collections contains -> This collection contains

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 9]
```diff
@@ -1,1267 +1,1228 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "ydqmdcDxCeXb"
+   },
+   "source": [
+    "In this tutorial, we shall first train these embeddings on speaker-related datasets, and then get speaker embeddings from a pretrained network for a new dataset. Since Google Colab has very slow read-write speeds, I'll be demonstrating this tutorial using [an4](http://www.speech.cs.cmu.edu/databases/an4/). \n",
+    "\n",
+    "Instead, if you'd like to try on a bigger dataset like [hi-mia](https://arxiv.org/abs/1912.01231) use the [get_hi-mia-data.py](https://github.com/NVIDIA/NeMo/blob/stable/scripts/dataset_processing/get_hi-mia_data.py) script to download the necessary files, extract them, also re-sample to 16Khz if any of these samples are not at 16Khz. "
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:61]: also re-sample -> and resample

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 10]
```diff
@@ -1,1267 +1,1228 @@
...
+   "metadata": {
+    "colab_type": "text",
+    "id": "oDzak_FIB9LS"
+   },
+   "source": [
+    "# **SPEAKER RECOGNITION** \n",
+    "Speaker Recognition (SR) is a broad research area that solves two major tasks: speaker identification (who is speaking?) and\n",
+    "speaker verification (is the speaker who they claim to be?). In this work, we focus on text-independent speaker recognition when the identity of the speaker is based on how the speech is spoken,\n",
+    "not necessarily in what is being said. Typically such SR systems operate on unconstrained speech utterances,\n",
+    "which are converted into vectors of fixed length, called speaker embeddings. Speaker embeddings are also used in\n",
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:48]: vectors of fixed length -> fixed-length vectors

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 11]
```diff
@@ -1,1267 +1,1228 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "m5Zho11LNAFJ"
+   },
+   "source": [
+    "## Model Configuration \n",
+    "The TitaNet Model is defined in a config file which declares multiple important sections.\n",
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:307]: TitaNet Model -> TitaNet model

**Review by tango4j — CHANGES_REQUESTED** [review]
Checked the tutorial running and found out some errors/suggestion points.

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb [hunk 12]
```diff
@@ -1,1267 +1,1228 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "VciRUIRz0y6P"
+   },
+   "source": [
+    "Training for a speaker verification model is almost the same as the speaker recognition model with a change in the loss function. Angular Loss is a better function to train for a speaker verification model as the model is trained in an end to end manner with loss optimizing for embeddings cluster to be far from each other for different speaker by maximizing the angle between these clusters"
```
Comment (tango4j) [inline tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb:996]: The words should be hyphenated if multiple words are used as an adjective:
end to end -> end-to-end

File: tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb (hunk 2 above)
Comment (tango4j) [reply tutorials/speaker_tasks/Speaker_Identification_Verification.ipynb]: end-to-end (hyphenated words)

**Review by tango4j — APPROVED** [review]
